geometry/interval.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+namespace geometry {
+namespace internal_interval {
+
+using quantities::Difference;
+using quantities::Infinity;
+
+// Represents the interval [min, max]; T must be an ordered affine space.
+template<typename T>
+struct Interval {
```

Comment:
I would prefer it if this was a class and `min` and `max` were selectors.  You should really not shoot directly at `min` and `max`.

Reply:
Maybe later.